tests/generic/mongocursor-tailable-003.phpt

```diff
@@ -0,0 +1,115 @@
+--TEST--
+MongoCursor::tailable() with getNext() and hasNext() (initial query matches one document)
+--SKIPIF--
+<?php if (getenv('SKIP_SLOW_TESTS')) die('skip slow tests excluded by request'); ?>
+<?php require_once "tests/utils/standalone.inc" ?>
+--FILE--
+<?php
+require_once "tests/utils/server.inc";
+
+function log_getmore($server, $info) {
+    //echo "Issuing getmore\n";
+}
+
+$ctx = stream_context_create(array(
+    'mongodb' => array(
+        'log_getmore' => 'log_getmore',
+    ),
+));
+
+$host = MongoShellServer::getStandaloneInfo();
+$m = new MongoClient($host, array(), array('context' => $ctx));
+
+$db = $m->selectDB(dbname());
+$db->selectCollection(collname(__FILE__))->drop();
+$db->createCollection(collname(__FILE__), array('capped' => true, 'size' => 10240, 'max' => 3));
+
+$coll = $db->selectCollection(collname(__FILE__));
+$cursor = $coll->find()->tailable();
+
+$x = 0;
+$i = 0;
+
+$doInsert = function() use ($coll, &$x) {
+    $coll->insert(array('x' => $x));
+    printf("Inserted: x=%d\n", $x);
+    ++$x;
+};
+
+$doInsert();
+
+while (++$i) {
+    printf("\nLoop iteration: $i\n", $i);
+
+    $hasNext = $cursor->hasNext();
+    printf("hasNext: %s\n", $hasNext ? 'true' : 'false');
+
+    if ($hasNext) {
+        $result = $cursor->getNext();
+        printf("Found: x=%d\n", $result['x']);
+        continue;
+    }
+
+    echo "Found nothing\n";
+
+    if ($cursor->dead()) {
+        echo "Cursor is dead; breaking loop\n";
+        break;
+    }
+
+    if ($x >= 5) {
+        echo "Stopping at x >= 5\n";
+        break;
+    }
+
+    $doInsert();
+    $doInsert();
+
+    echo "Sleeping for one second\n";
+    sleep(1);
+}
+
+?>
+===DONE===
+--EXPECTF--
+Inserted: x=0
+
+Loop iteration: 1
```

Comment:
In case the expected output and code is not straightforward, this is what I'm trying to test:

 * Each loop iteration, we attempt to find a single document using `hasNext()` and `getNext()`. We'll always debug `hasNext()` and print the document if we do match it. On any match, we'll jump to the next iteration and repeat.
 * If we didn't find anything, we'll insert a couple of documents and sleep. This gives us more "food" for the next consumption. We sleep because we're not using `awaitData()` and I'd like to pause before tailing again.
 * If we found nothing we have two chances to abort our infinite loop:
   * The cursor is dead
   * `$x >= 5`, which means we've already inserted 5 documents.